### DISCUSSION TITLE
Otel Collector based on a graph database?

### DISCUSSION THREAD

**Discussion by RichMorin** [open-telemetry/opentelemetry-erlang#662] [Ideas]
I've been speculating about creating an [Otel Collector](https://opentelemetry.io/docs/collector/) based on a [graph database](https://en.wikipedia.org/wiki/Graph_database) (e.g., [ArangoDB](https://en.wikipedia.org/wiki/ArangoDB), [Neo4j](https://en.wikipedia.org/wiki/Neo4j)). This could collect data from other Otel sources, store and index it using the database, and then support queries from members of the Otel ecosystem. Ideally, I'd like this to be implemented in Elixir. And a pony...

So, can anyone give me an idea of how close the existing infrastructure (in opentelemetry-erlang, opentelemetry-*, et al) is to making this possible? Better yet, what might the Golden Path be to creating this sort of resource?

**Comment by bryannaegele** [comment]
It sounds like you want to build some entirely new tool that intakes otlp and does stuff with it. I don't know how much the erlang/elixir API and SDK would be needed or usable for you aside from possibly taking the exporter protobuf encoding code and reverse-engineer that. The collector is just an optional pit stop for doing things with the data coming in before exporting it to another destination, like the tool you're describing.

There is an #otel-collector channel in the CNCF Slack that could be a good spot to get some input.
